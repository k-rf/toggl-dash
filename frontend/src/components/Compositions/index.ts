export * from "./Form";
export * from "./RhfTextField";
