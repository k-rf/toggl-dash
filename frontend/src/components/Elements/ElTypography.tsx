import { Typography, TypographyProps } from "@mui/material";

type Props = Pick<TypographyProps, "children" | "variant" | "color" | "sx">;

export const ElTypography = (props: Props) => {
  return <Typography {...props} />;
};
