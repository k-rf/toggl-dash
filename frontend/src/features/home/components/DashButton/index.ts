export * from "./DashButton";
export * from "./DashButtonDetailListItem";
