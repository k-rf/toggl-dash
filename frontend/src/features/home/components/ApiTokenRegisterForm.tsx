import { InputAdornment, Stack } from "@mui/material";
import { KeyboardEventHandler, MouseEventHandler, useState } from "react";

import { Form, HandleSubmit, Methods } from "~/components/Compositions";
import { ElButton, ElTextField } from "~/components/Elements";
import { ElIconButton } from "~/components/Elements/ElIconButton";
import { ElTypography } from "~/components/Elements/ElTypography";
import { VisibilityIcon, VisibilityOffIcon } from "~/components/Icons";
import { z } from "~/lib/zod";

const schema = z.object({
  apiToken: z.string().min(1),
});

type ApiTokenValues = z.infer<typeof schema>;

type Props = { onSubmit?: (value: ApiTokenValues) => Promise<void> | void };

export const ApiTokenRegisterForm = (props: Props) => {
  const [visible, setVisible] = useState(false);

  const submit = (methods: Methods<ApiTokenValues>) => {
    return methods.handleSubmit(async (value) => {
      methods.reset({ apiToken: "" });

      await props.onSubmit?.(value);
    });
  };

  const handleSubmit: HandleSubmit<ApiTokenValues, MouseEventHandler> = (methods) => {
    return submit(methods);
  };

  const handleKeyDown: HandleSubmit<ApiTokenValues, KeyboardEventHandler> = (methods) => {
    return (e) => {
      if (e.code === "Enter") {
        e.preventDefault();

        if (e.ctrlKey) {
          return submit(methods)(e);
        }
      }
    };
  };

  const handleShow = () => {
    setVisible(true);
  };

  const handleMask = () => {
    setVisible(false);
  };

  return (
    <Form<ApiTokenValues> options={{ mode: "all" }} schema={schema}>
      {(methods) => {
        const { error } = methods.getFieldState("apiToken");
        const {
          errors: { apiToken },
          isValid,
        } = methods.formState;

        return (
          <Stack spacing={1}>
            <ElTypography>Toggl の API トークンを入力してください</ElTypography>
            <ElTextField
              fullWidth
              placeholder="API トークン"
              type={visible ? "text" : "password"}
              error={Boolean(error)}
              onKeyDown={handleKeyDown(methods)}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <ElIconButton
                      disableTouchRipple
                      onMouseDown={handleShow}
                      onMouseUp={handleMask}
                      onMouseLeave={handleMask}
                      data-testid="api-token-register-form-visibility-toggle-icon-button"
                    >
                      {visible ? <VisibilityOffIcon /> : <VisibilityIcon />}
                    </ElIconButton>
                  </InputAdornment>
                ),
              }}
              registration={methods.register("apiToken")}
            />
            <ElButton
              disabled={!isValid}
              onClick={handleSubmit(methods)}
              data-testid="api-token-register-form-submit-button"
            >
              保存する
            </ElButton>
            <ElTypography variant="body2" color="error">
              {apiToken?.message}
            </ElTypography>
          </Stack>
        );
      }}
    </Form>
  );
};

export type { Props as ApiTokenRegisterFormProps };
