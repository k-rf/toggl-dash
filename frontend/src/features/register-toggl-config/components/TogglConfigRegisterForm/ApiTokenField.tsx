import { InputAdornment } from "@mui/material";
import { KeyboardEventHandler, useState } from "react";
import { UseFormRegisterReturn, FieldError } from "react-hook-form";

import { ElTextField } from "~/components/Elements";
import { ElIconButton } from "~/components/Elements/ElIconButton";
import { VisibilityOffIcon, VisibilityIcon } from "~/components/Icons";

type Props = {
  error?: FieldError;
  onKeyDown: KeyboardEventHandler<Element>;
  registration: Partial<UseFormRegisterReturn>;
};

export const ApiTokenField = (props: Props) => {
  const [visible, setVisible] = useState(false);

  const handleShow = () => {
    setVisible(true);
  };

  const handleMask = () => {
    setVisible(false);
  };

  return (
    <ElTextField
      id="api-token-field"
      fullWidth
      size="small"
      placeholder="API トークン"
      type={visible ? "text" : "password"}
      error={Boolean(props.error)}
      onKeyDown={props.onKeyDown}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">
            <ElIconButton
              disableTouchRipple
              onMouseDown={handleShow}
              onMouseUp={handleMask}
              onMouseLeave={handleMask}
              data-testid="api-token-register-form-visibility-toggle-icon-button"
            >
              {visible ? <VisibilityOffIcon /> : <VisibilityIcon />}
            </ElIconButton>
          </InputAdornment>
        ),
      }}
      registration={props.registration}
    />
  );
};
