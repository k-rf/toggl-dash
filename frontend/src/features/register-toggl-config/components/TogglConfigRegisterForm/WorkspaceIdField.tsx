import { KeyboardEventHandler } from "react";
import { FieldError, UseFormRegisterReturn } from "react-hook-form";

import { ElTextField } from "~/components/Elements";

type Props = {
  error?: FieldError;
  onKeyDown: KeyboardEventHandler<Element>;
  registration: Partial<UseFormRegisterReturn>;
};

export const WorkspaceIdField = (props: Props) => {
  return (
    <ElTextField
      id="workspace-id-field"
      fullWidth
      size="small"
      placeholder="ワークスペース ID"
      type="text"
      error={Boolean(props.error)}
      onKeyDown={props.onKeyDown}
      registration={props.registration}
    />
  );
};
