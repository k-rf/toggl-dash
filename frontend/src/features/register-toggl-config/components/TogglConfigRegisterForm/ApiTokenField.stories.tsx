import type { Meta, StoryObj } from "@storybook/react";

import { ApiTokenField, ApiTokenFieldProps } from "./ApiTokenField";

export default { component: ApiTokenField } as Meta<ApiTokenFieldProps>;

type Story = StoryObj<ApiTokenFieldProps>;

export const Default: Story = {};
