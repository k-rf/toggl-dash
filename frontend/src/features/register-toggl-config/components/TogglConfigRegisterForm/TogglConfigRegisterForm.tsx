import { Stack } from "@mui/material";
import { KeyboardEventHandler, MouseEventHandler } from "react";

import { Form, HandleSubmit, Methods } from "~/components/Compositions";
import { ElButton } from "~/components/Elements";
import { ElTypography } from "~/components/Elements/ElTypography";
import { z } from "~/lib/zod";

import { ApiTokenField } from "./ApiTokenField";

const schema = z.object({
  apiToken: z.string().min(1),
});

type ApiTokenValues = z.infer<typeof schema>;

type Props = { onSubmit?: (value: ApiTokenValues) => Promise<void> | void };

export const TogglConfigRegisterForm = (props: Props) => {
  const submit = (methods: Methods<ApiTokenValues>) => {
    return methods.handleSubmit(async (value) => {
      methods.reset({ apiToken: "" });

      await props.onSubmit?.(value);
    });
  };

  const handleKeyDown: HandleSubmit<ApiTokenValues, KeyboardEventHandler> = (methods) => {
    return (e) => {
      if (e.code === "Enter") {
        e.preventDefault();

        if (e.ctrlKey) {
          return submit(methods)(e);
        }
      }
    };
  };

  const handleSubmit: HandleSubmit<ApiTokenValues, MouseEventHandler> = (methods) => {
    return submit(methods);
  };

  return (
    <Form<ApiTokenValues> options={{ mode: "all" }} schema={schema}>
      {(methods) => {
        const { error } = methods.getFieldState("apiToken");
        const {
          errors: { apiToken: apiTokenError },
          isValid,
        } = methods.formState;

        return (
          <Stack spacing={1}>
            <ApiTokenField
              error={error}
              onKeyDown={handleKeyDown(methods)}
              registration={methods.register("apiToken")}
            />
            <ElButton
              disabled={!isValid}
              onClick={handleSubmit(methods)}
              data-testid="api-token-register-form-submit-button"
            >
              保存する
            </ElButton>
            <ElTypography variant="body2" color="error" sx={{ height: "max-content" }}>
              {apiTokenError?.message}
            </ElTypography>
          </Stack>
        );
      }}
    </Form>
  );
};

export type { Props as ApiTokenRegisterFormProps };
