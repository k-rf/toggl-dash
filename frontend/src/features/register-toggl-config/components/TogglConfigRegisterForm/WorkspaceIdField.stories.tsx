import type { Meta, StoryObj } from "@storybook/react";

import { WorkspaceIdField } from "./WorkspaceIdField";

export default { component: WorkspaceIdField } as Meta<typeof WorkspaceIdField>;

type Story = StoryObj<typeof WorkspaceIdField>;

export const Default: Story = {};
