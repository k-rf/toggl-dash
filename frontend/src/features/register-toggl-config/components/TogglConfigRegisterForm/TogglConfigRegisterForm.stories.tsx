import { expect } from "@storybook/jest";
import type { Meta, StoryObj } from "@storybook/react";
import { fireEvent, userEvent, waitFor, within } from "@storybook/testing-library";

import { TogglConfigRegisterForm, ApiTokenRegisterFormProps } from "./TogglConfigRegisterForm";

export default { component: TogglConfigRegisterForm } as Meta<ApiTokenRegisterFormProps>;

type Story = StoryObj<ApiTokenRegisterFormProps>;

export const Default: Story = {
  argTypes: { onSubmit: { description: "Submit form data" } },
};

export const Initial: Story = {
  ...Default,
  name: "初期状態のとき、何も入力されておらず保存ボタンも非活性状態になっている",
  play: playUtils(async (ctx) => {
    await expect(ctx.tokenField).toHaveValue("");
    await expect(ctx.submitButton).toBeDisabled();
  }),
};

export const Submit: Story = {
  ...Default,
  name: "API トークンを入力し保存ボタンを押下すると、入力内容がリセットされる",
  play: playUtils(async (ctx) => {
    await ctx.inputTokenField("token");
    await ctx.clickSubmitButton();

    await waitFor(async () => {
      await expect(ctx.tokenField).toHaveValue("");
    });
  }),
};

export const NoInput: Story = {
  ...Default,
  name: "トークンを入力せずにフォーカスアウトすると、エラーメッセージを表示する",
  play: playUtils(async (ctx) => {
    await fireEvent.focusIn(ctx.tokenField);
    await fireEvent.focusOut(ctx.tokenField);

    await waitFor(async () => {
      await expect(await ctx.errorMessage()).toBeInTheDocument();
    });
  }),
};

type Context = {
  submitButton: HTMLElement;
  tokenField: HTMLElement;
  visibilityIcon: HTMLElement;
  errorMessage: () => Promise<HTMLElement>;
  inputTokenField: (data: string) => Promise<void>;
  clickSubmitButton: () => Promise<void>;
};

function playUtils(play: (context: Context) => Promise<void> | void) {
  return async ({ canvasElement }: { canvasElement: HTMLElement }) => {
    const tokenField = within(canvasElement).getByPlaceholderText("API トークン");
    const submitButton = within(canvasElement).getByTestId("api-token-register-form-submit-button");
    const visibilityIcon = within(canvasElement).getByTestId(
      "api-token-register-form-visibility-toggle-icon-button"
    );
    const errorMessage = async () => await within(canvasElement).findByText("必須項目です");

    return await play({
      tokenField,
      submitButton,
      errorMessage,
      visibilityIcon,
      inputTokenField: async (data: string) => {
        await userEvent.type(tokenField, data);
      },
      clickSubmitButton: async () => {
        await waitFor(async () => {
          await expect(submitButton).toBeEnabled();
        });

        await userEvent.click(submitButton);
      },
    });
  };
}
