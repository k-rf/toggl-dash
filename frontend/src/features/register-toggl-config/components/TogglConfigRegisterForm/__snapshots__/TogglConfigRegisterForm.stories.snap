// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features/Register-toggl-config/Components/TogglConfigRegisterForm/TogglConfigRegisterForm > API トークンを入力し保存ボタンを押下すると、入力内容がリセットされる 1`] = `
<form>
  <div
    className="css-1okj3ks-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r4:"
          name="apiToken"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="API トークン"
          required={false}
          type="password"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="api-token-register-form-visibility-toggle-icon-button"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1t4o62d-MuiButtonBase-root-MuiButton-root"
      data-testid="api-token-register-form-submit-button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      保存する
    </button>
    <p
      className="MuiTypography-root MuiTypography-body2 css-coaz7i-MuiTypography-root"
    />
  </div>
</form>
`;

exports[`Features/Register-toggl-config/Components/TogglConfigRegisterForm/TogglConfigRegisterForm > Default 1`] = `
<form>
  <div
    className="css-1okj3ks-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r2:"
          name="apiToken"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="API トークン"
          required={false}
          type="password"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="api-token-register-form-visibility-toggle-icon-button"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1t4o62d-MuiButtonBase-root-MuiButton-root"
      data-testid="api-token-register-form-submit-button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      保存する
    </button>
    <p
      className="MuiTypography-root MuiTypography-body2 css-coaz7i-MuiTypography-root"
    />
  </div>
</form>
`;

exports[`Features/Register-toggl-config/Components/TogglConfigRegisterForm/TogglConfigRegisterForm > トークンを入力せずにフォーカスアウトすると、エラーメッセージを表示する 1`] = `
<form>
  <div
    className="css-1okj3ks-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r5:"
          name="apiToken"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="API トークン"
          required={false}
          type="password"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="api-token-register-form-visibility-toggle-icon-button"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1t4o62d-MuiButtonBase-root-MuiButton-root"
      data-testid="api-token-register-form-submit-button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      保存する
    </button>
    <p
      className="MuiTypography-root MuiTypography-body2 css-coaz7i-MuiTypography-root"
    />
  </div>
</form>
`;

exports[`Features/Register-toggl-config/Components/TogglConfigRegisterForm/TogglConfigRegisterForm > 初期状態のとき、何も入力されておらず保存ボタンも非活性状態になっている 1`] = `
<form>
  <div
    className="css-1okj3ks-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r3:"
          name="apiToken"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="API トークン"
          required={false}
          type="password"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="api-token-register-form-visibility-toggle-icon-button"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1t4o62d-MuiButtonBase-root-MuiButton-root"
      data-testid="api-token-register-form-submit-button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      保存する
    </button>
    <p
      className="MuiTypography-root MuiTypography-body2 css-coaz7i-MuiTypography-root"
    />
  </div>
</form>
`;
